src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -526,7 +526,7 @@ public static CommandLineBuilder UseHelp(this CommandLineBuilder builder)
                 description: Resources.Instance.VersionOptionDescription(),
                 parseArgument: result =>
                 {
-                    if (result.FindResultFor(command)?.Children.Count > 1)
+                    if (result.FindResultFor(command)?.Children.Where(IsNotImplicit).Count() > 1)
```

Comment:
Can we do this without the LINQ? A bunch of LINQ removal was done for performance reasons.

I wonder if a simpler implementation could work based on just the token count?